src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,274 @@
+                {
+                    continue;
+                }
+
+                var currentNode = node;
+
+                if (needsRename)
+                {
+                    currentNode = ((SimpleNameSyntax)currentNode).WithIdentifier(identifierToken);
+                }
+
+                if (hasAdditionalTypeArguments)
+                {
+                    var existingTypeArguments = symbol.TypeArguments.Select(x => x.GenerateTypeSyntax());
+                    // Prepend additional type arguments to preserve lexical order in which they are defined
+                    var typeArguments = additionalTypeArguments.Concat(existingTypeArguments);
+                    currentNode = generator.WithTypeArguments(currentNode, typeArguments);
+                }
+
+                if (node.Parent.IsKind(SyntaxKind.InvocationExpression, out InvocationExpressionSyntax invocation))
+                {
+                    if (hasAdditionalArguments)
+                    {
+                        var shouldUseNamedArguments =
+                            !supportsNonTrailing && invocation.ArgumentList.Arguments.Any(arg => arg.NameColon != null);
+
+                        var additionalArguments = capturesAsParameters.Select(parameter =>
+                            (ArgumentSyntax)generator.Argument(
+                                name: shouldUseNamedArguments ? parameter.Name : null,
+                                refKind: parameter.RefKind,
+                                expression: parameter.Name.ToIdentifierName())).ToArray();
+
+                        editor.ReplaceNode(invocation.ArgumentList,
+                            invocation.ArgumentList.AddArguments(additionalArguments));
+                    }
+                }
+                else if (hasAdditionalArguments || hasAdditionalTypeArguments)
+                {
+                    // Convert local function delegates to lambda if the signature no longer matches
+                    var arguments = parameters
+                        .Concat(capturesAsParameters)
+                        .Select(parameter => generator.Argument(
+                            name: null, refKind: parameter.RefKind,
+                            expression: parameter.Name.ToIdentifierName()));
+
+                    currentNode = generator.ValueReturningLambdaExpression(
+                        lambdaParameters: ParameterGenerator.GetParameters(parameters, isExplicit: true, options: defaultOptions),
+                        expression: generator.InvocationExpression(currentNode, arguments));
+                }
+
+                editor.ReplaceNode(node, currentNode);
+            }
+
+            root = editor.GetChangedRoot();
+            localFunction = (LocalFunctionStatementSyntax)root.FindNode(localFunction.Span);
+
+            method = WithBodyFrom(method, localFunction);
+            editor = new SyntaxEditor(root, generator);
```

Comment:
had to recreate syntax editor to replace updated function body. Is there a better way?

Reply:
that's fine.  it's not uncommon to use multiple editors.